src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -406,6 +406,19 @@ public string GetDebuggerDisplay()
         // For purpose of nullability analysis, awaits create pending branches, so async usings and foreachs do too
         public sealed override bool AwaitUsingAndForeachAddsPendingBranch => true;
 
+        protected override void EnsureSufficientExecutionStack(int recursionDepth)
+        {
+            if (recursionDepth > StackGuard.MaxUncheckedRecursionDepth &&
+                compilation.NullableAnalysisData is { MaxRecursionDepth: var depth } &&
+                depth > 0 &&
```

Comment:
>depth > 0 && [](start = 16, length = 12)

Is this check necessary?

Reply:
Yes, the default value is `-1` which is intended to mean no limit.

---
In reply to: [595301185](https://github.com/dotnet/roslyn/pull/51851#discussion_r595301185) [](ancestors = 595301185)